packages/web-components/fast-foundation/src/dialog/dialog.ts

```diff
@@ -96,6 +96,8 @@ export class Dialog extends FoundationElement {
      */
     public dismiss(): void {
         this.$emit("dismiss");
```

Comment:
@bennypowers Thanks for this! @EisenbergEffect thoughts on how we might *deprecate* the `dismiss` event on line 98? If we can mark it deprecated somehow we can add a comment above to remove in the next major version.

Reply:
sounds like a good use for custom-elements.json

Reply:
@bennypowers Does the custom-elements.json have a way to indicate that something is deprecated? If not, perhaps we just leave the event in there for now with a comment that it's deprecated, and then remove any documentation on this either on our site or the spec. Then, we can stop publishing the event in a future version.

Reply:
Looking more at the code, I'm inclined to deprecate the `dismiss` and `hide` methods, keeping their old event and state behaviors in place. Then add new `cancel()` and `close()` methods with the improved event patterns.

Reply:
there are two options in custom-elements.json
1) use a non-standard field and consume it in your own tooling. I opened https://github.com/webcomponents/custom-elements-manifest/issues/84 to track the idea in the spec
2) Prepend `**DEPRECATED**` to description members

`HTMLDialogElement` has `close()`, `show()`, and `showModal()`

https://developer.mozilla.org/en-US/docs/Web/API/HTMLDialogElement

Reply:
@bennypowers @EisenbergEffect @chrisdholt how are we feeling about this PR? Are there changes that still need to be made? As a stale Draft status it seems we should either remove Draft and work to get it updated to suit the comments or abandon?  Thoughts?

Reply:
I'd love to have this still come in, however I know that @bennypowers was running into build issues that we weren't able to repro. Definitely happy to find a way to help get beyond that if the issue still persists, but if I recall it was quite frustrating and led to this sitting in the ether.

Reply:
Happy to take another crack at this

Reply:
Maintainers should please permit CI to run

Reply:
> @bennypowers Thanks for this! @EisenbergEffect thoughts on how we might _deprecate_ the `dismiss` event on line 98? If we can mark it deprecated somehow we can add a comment above to remove in the next major version.

Also, since this thread started, [`deprecated` got added to the custom-elements manifest schema](https://github.com/webcomponents/custom-elements-manifest/pull/89)

Reply:
@williamw2 Since you are investigating custom elements manifest, it would be great if we could support the new `deprecated` feature. It might be good to sync up with @nicholasrice on some of this as well. I think there's a possibility of using the API extractor metadata from our code to generate the custom elements manifest. There was recent work which Nick contributed to, to enable custom documentation tags. I *think* it's all merged and ready for use now. If we can use that, we can add custom tags for slots, deprecated, etc. and then use the API extractor metadata to generate the manifest from our code.

Reply:
Yep, docs for that feature are here: https://api-extractor.com/pages/configs/tsdoc_json/